tests/unit_tests/agents/conversational_chat/test_output_parser.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+from langchain.agents import AgentType, initialize_agent
+from langchain.chat_models import ChatOpenAI
+from langchain.memory import ConversationBufferWindowMemory
+
+# set no tools available (generate only)
+notoolset = []
```

Comment:
Something along these lines for the file:

```python

from __future__ import annotations

import pytest

from langchain.agents.conversational_chat.output_parser import ConvoOutputParser, OutputParserException


@pytest.mark.parametrize(
    "text, expected",
    [
        ('text', (expected value) ),

    ]
)
def test_parser(text, expected) -> None:
    """Test the ChatOutputParser class."""
    parser = ConvoOutputParser()
    assert parser.parse(text) == expected
    
@pytest.mark.parametrize(
    "text",
    [
        ('texts that lead to parse exception'),

    ]
)
def test_unhappy_path(text) -> None:
    """Test the ChatOutputParser class."""
    parser = ConvoOutputParser()
    with pytest.raises(OutputParserException):
        parser.parse(text)
```

Reply:
Very helpful. Thank you!

The unit tester has been rewritten to account for code and non-code return strings.